### ISSUE TITLE
Client application secrets not renewed on cluster reinstallation

### ISSUE THREAD

**Issue by dweber019** [strimzi/strimzi-kafka-operator#3376]
**Describe the bug**
Currently we are testing the restore procedure of our Strimzi installation. As we have set the deleteClaim to false the PVC should remain. We then removed the cluster and reinstalled it. Good to know the data are still there :wink: but what we found is that the client application secrets (keystore) are not recreated. This lead to a error as the CA certificates have been renewed and don't match anymore. We have then tried to restart the entity operator to trigger a recreation of the client application secrets but didn't do the trick.
In the documentation of renewing certificates (https://strimzi.io/docs/operators/master/using.html#renewal_process_with_generated_cas) there will be new certificates generated but apparently not for this case?

**To Reproduce**
1. Setup a new kafka cluster with deleteClaim: false
2. Create so KafkaTopic's and KafkaUser's
3. Undeploy the cluster (this should not remove the KafkaTopic's and KafkaUser's and PVC's)
4. Redeploy the cluster (this should only renew the CA certificates but not the client ones)

**Expected behavior**
Client application secrets will be renewed with the new CA certificates

**Environment (please complete the following information):**
 - Strimzi version: AMQ steams 1.4.1 = Strimzi 0.17.x
 - Installation method: YAML files
 - Kubernetes cluster: OpenShift 3.9

**Additional context**

**Comment by scholzj** [comment]
I tried to follow your reproducer with the 0.19.0-rc1. Bu I cannot reproduce it. I did following:
* Deployed Kafka cluster
* Created KafkaUser
* Made back up of the secret (for comparing the certs later)
* Deleted the Kafka cluster
    * This leaves as expected the KafkaUser and the secret intact
* I deploy new Kafka cluster with the same name
    * This uses new CAs since the old secrets were deleted => expected
* When the User Operator starts it automatically refresh the secret with new certificate signed by the new CA

I know I tried it with newer version. But I'm not aware of any changes to the related functionality since 0.17.0. So I'm not sure I would expect it to behave differently. But in case it was, it seems to be fixed now in the upcoming 0.19.0 release.

**Comment by dweber019** [comment]
I just retested on our cluster and did exactly the same and the certificates aren't recreated/updated. At least not with AMQ Streams 1.4.1.

The entity operator shows some logs that he needs to update the secrets but never does.
```
[2020-07-27 09:38:51,411] INFO  <UserOperator:72> [oop-thread-0] Triggering periodic reconciliation for namespace kafka-test...
[2020-07-27 09:38:51,468] INFO  <ractOperator:122> [oop-thread-0] Reconciliation #26(timer) User(kafka-test/kafka-user): User kafka-user should be created or updated
[2020-07-27 09:38:52,203] INFO  <ractOperator:262> [oop-thread-0] Reconciliation #26(timer) User(kafka-test/kafka-user): reconciled
```

**Comment by scholzj** [comment]
So ... is it your production cluster? If not, can you share the clients CA and the user secret? I will try it with Strimzi 0.17 to see whether there is something in that release.

**Comment by scholzj** [comment]
I tried it once again with Strimzi 0.17.0, but still nothing. The secret is properly refreshed. The code is fairly simple and I do not think there are too many options where it can go wrong. Could you share the secrets as I mentioned above + your Kafka CR to see if there isn't something wrong?

**Comment by dweber019** [comment]
I can't share the secrets but the CR's I can.
KafkaCluster.yaml (I have removed the metrics and kafka exporter)
```
apiVersion: kafka.strimzi.io/v1beta1
kind: Kafka
metadata:
  name: kafka
spec:
  kafka:
    version:  2.4.0
    template:
      pod:
        affinity:
            podAntiAffinity:
              requiredDuringSchedulingIgnoredDuringExecution:
                - labelSelector:
                    matchExpressions:
                      - key: kafkaAffinity
                        operator: In
                        values:
                          - brokerAffinity
                  topologyKey: kubernetes.io/hostname
      statefulset:
        metadata:
          labels:
            kafkaAffinity: brokerAffinity
    replicas: 3
    listeners:
      external:
        type: route
        authentication:
          type: tls
        overrides:
          bootstrap:
            host: kafka-kafka-bootstrap.kafka-test.app-test.example.com
          brokers:
            - broker: 0
              host: kafka-0-kafka-test.app-test.example.com
            - broker: 1
              host: kafka-1-kafka-test.app-test.example.com
            - broker: 2
              host: kafka-2-kafka-test.app-test.example.com
      tls:
        authentication:
          type: tls
    authorization:
      type: simple
    resources:
      requests:
        cpu: 25m
        memory: 1Gi
      limits:
        cpu: 1
        memory: 2Gi
    tlsSidecar:
      resources:
        requests:
          cpu: 75m
          memory: 64Mi
        limits:
          cpu: 75m
          memory: 64Mi
    config:
      offsets.topic.replication.factor: 3
      transaction.state.log.replication.factor: 3
      transaction.state.log.min.isr: 2
      min.insync.replicas: 2
      default.replication.factor: 3
      log.message.format.version: "2.4"
      auto.create.topics.enable: true
    storage:
      type: jbod
      volumes:
        - id: 0
          type: persistent-claim
          size: 5Gi
          deleteClaim: false
          class: trident-prod-backup
    jvmOptions:
      gcLoggingEnabled: false
      -Xms: 500m
      -Xmx: 500m
  zookeeper:
    template:
      pod:
        affinity:
              podAntiAffinity:
                requiredDuringSchedulingIgnoredDuringExecution:
                  - labelSelector:
                      matchExpressions:
                        - key: kafkaAffinity
                          operator: In
                          values:
                            - zookeeperAffinity
                    topologyKey: kubernetes.io/hostname
      statefulset:
        metadata:
          labels:
            kafkaAffinity: zookeeperAffinity
    replicas: 3
    storage:
      type: persistent-claim
      size: 5Gi
      deleteClaim: false
      class: trident-prod-backup
    resources:
      requests:
        cpu: 25m
        memory: 512Mi
      limits:
        cpu: 500m
        memory: 1Gi
    tlsSidecar:
      resources:
        requests:
          cpu: 75m
          memory: 64Mi
        limits:
          cpu: 75m
          memory: 64Mi
    jvmOptions:
      gcLoggingEnabled: false
  entityOperator:
    topicOperator:
      resources:
        requests:
          cpu: 50m
          memory: 256Mi
        limits:
          cpu: 750m
          memory: 512Mi
    userOperator:
      resources:
        requests:
          cpu: 50m
          memory: 256Mi
        limits:
          cpu: 750m
          memory: 512Mi
    tlsSidecar:
      resources:
        requests:
          cpu: 75m
          memory: 64Mi
        limits:
          cpu: 75m
          memory: 64Mi
```

**Comment by scholzj** [comment]
I tried to reproduce it again with the Kafka CR more similar to yours. But it still works fine for me and I do not see anything what would break it in the CR. I'm afraid that without the secrets there is not much way to move forward. The User operator is comparing the Clients CA with the CA in the user secret and if they don't match it will generate a new key. So I have no idea how could your Client CA be different and yet equal to the old one.

**Comment by borj** [comment]
"... We have then tried to restart the entity operator to trigger a recreation of the client application secrets but didn't do the trick..."

Did you manage to solve this? We are in a similar situation, EO certs just wont renew. We forced the CA regeneration (as per the documentation, with annotation etc) in another environment (that is functioning properly) and everything went smooth and as expected. In the important one we had expired certs, ran the procedure, zookeepers, kafkas and EO were restarted by the process, all the client and cluster CA certs seem to be new (they are in fact) but the EO certificates just wont renew. So EO is crashing, other components can not communicate etc...
